### PR TITLE
create button component

### LINKED ISSUE #4: Button Component
## Expected Behaviour

Create a button that has 2 variants (primary, secondary) and 2 sizes (default, small)
A button can be also disabled
The button can take text to display

## Checklist

_Detail the steps below to help us reproduce the problem you encountered._

- [ ] Follow the component design in figma https://www.figma.com/design/TOCJzJFMXE7PL1dQyGWqXZ/midterm?node-id=193-51&t=3bsn1O4cipxo3ue9-1
- [ ] Define necessary props

### PR DESCRIPTION
**Pull request by jsf030**
## Checklist

Confirm you have completed the following actions prior to submitting this PR.

- [x] There is an existing issue report for this PR.
- [x] I have created a feature branch.
- [x] My changes have been committed.
- [x] I have pushed my changes to the branch.

## Title

_Give your PR a short title summarising the patch, bug fix or feature._

## Description

_Ensure the PR description clearly describes the problem and solution and provide as much relevant information as possible._

## Issue Resolution

_Tell us which issue this PR fixes._

Closes #4 

## Proposed Changes

_List your proposed changes below._

-
-
-

## New or Changed Features

_Does this PR provide new or changed features or enhancements? If so, what is included in this PR?_

-
-
-

### CODE REVIEW

**Comment by dnmct** [comment]
hallo

File: src/components/button.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+export function Button({
+  children,
+  variant = "primary",
+  size = "default",
+}: Props) {
+  return (
+    <button
+      className={cn(
+        "rounded-lg font-semibold",
+        "hover:shadow-xl",
```
Comment (dnmct) [inline src/components/button.tsx]: we don't show a shadow on hover

File: src/components/button.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        variant === "primary" ? "bg-yellow disabled:bg-yellow/50" : null,
+        variant === "secondary" ? "bg-dark-light" : null,
+        size === "default" ? "py-4 w-full" : null,
+        size === "small" ? "py-2 w-full" : null
```
Comment (dnmct) [inline src/components/button.tsx]: - padding is not correct for small, 
- also text should be smaller on small size
- w-full is a base style

**Review by dnmct — CHANGES_REQUESTED** [review]

File: src/components/button.tsx (hunk 1 above)
Comment (jsf030) [reply src/components/button.tsx]: Ja, das und Ghost hatte ich auch rausoperiert und es war dann doch wieder drin. :O

**Review by dnmct — APPROVED** [review]
Good work 👍🏻

**Pull request merged**